### PR TITLE
[rework] third person camera view

### LINKED ISSUE #233: [Rework] Camera mechanics
- animasi player ngikutin angle kamera (statis ke arah player, relatif terhadap kamera)

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- rework Player Follow camera
- rework player animation system

### Tasks List
- [close #233 ]

### Attachments

https://user-images.githubusercontent.com/51603162/157784643-b373392a-00da-49b5-a37c-e63d9d5f50df.mp4

### CODE REVIEW

**Review by JahnsMichael — COMMENTED** [review]
I think the camera rotation is too sensitive, wdyt? Here is the recording for comparison between how much the camera was rotated and how much my mouse pointer was moved.

https://user-images.githubusercontent.com/60375650/157816313-bee91cef-f046-48ab-868b-6550eb72aee5.mp4

**Review by naufalhilmi12345 — APPROVED** [review]
Its good for mee! Cool

File: Assets/Astralization/Scripts/Player/PlayerStates/HidingState.cs [hunk 1]
```diff
@@ -57,25 +57,15 @@ private void PlayerMovementChangeState()
...
 
     #region InputHandler
     public override void UnhidePlayer(InputAction.CallbackContext ctx)
     {
         if (ctx.performed)
         {
-            SendStopHidingHudEvent();
-            StartCoroutine(Utils.DelayerHelper.Delay(1.0f, SendStopHidingEvent));
+            StopHidingHudEvent?.Invoke(false);
+            StartCoroutine(Utils.DelayerHelper.Delay(1.0f, () => StopHidingEvent?.Invoke()));
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Player/PlayerStates/HidingState.cs:68]: Make use owner.changestate

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
- Adjust line endings: UiHelper.cs
- Make Utils subfolders (Calculation, Helpers, etc.) Naming is up to you
- Delete init test scene

**Review by raihansyahdean — APPROVED** [review]
Nice camera rework

**Pull request merged**